source/h1_encoder.c

```diff
@@ -181,6 +181,57 @@ static void s_write_headers(struct aws_byte_buf *dst, const struct aws_http_mess
     AWS_ASSERT(wrote_all);
 }
 
+static int s_headers_size(const struct aws_http_headers *headers, size_t *out_size) {
```

Comment:
I have nit-picky feelings about how we can share this code for trailers and traditional headers but it would take too long to type out, let's just talk tomorrow